src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageView.axaml.cs

```diff
@@ -16,8 +23,37 @@ public FileOriginsPageView()
                     vm => vm.FileOrigins,
                     v => v.DataGrid.ItemsSource)
                 .DisposeWith(d);
-            
+
+            DataGrid.SelectedItemsToViewModelObservableChangeSetProperty(ViewModel!, vm => vm.SelectedModsObservable)
+                .DisposeWith(d);
+
+            // Synchronize the Grid with DataModel for Selected Items
+            // Enable/Disable Add Mod & Add Mod Advanced Buttons
+            void UpdateAddButtonState(bool isModAdded)
+            {
+                var enable = false;
+                if (!isModAdded)
+                    enable = ViewModel!.SelectedModsCollection.Count > 0 && 
+                             !ViewModel!.SelectedModsCollection.Any(x => x.IsModAddedToLoadout);
+
+                AddModButton.IsEnabled = enable;
+                // Note Add(Advanced) should always be available.
+            }
+
+            ViewModel!.SelectedModsCollection.ObserveCollectionChanges()
+                .Subscribe(_ => { UpdateAddButtonState(false); })
+                .DisposeWith(d);
+            ViewModel!.SelectedModsObservable.WhenValueChanged(model => model.IsModAddedToLoadout)
+                .Subscribe(UpdateAddButtonState)
+                .DisposeWith(d);
+
             DataGrid.Width = Double.NaN;
+            
+            AddModButton.Command = ReactiveCommand.CreateFromTask(ViewModel!.AddMod);
+            AddModAdvancedButton.Command = ReactiveCommand.CreateFromTask(ViewModel!.AddModAdvanced);
+            GetModsFromNexusButton.Command = ReactiveCommand.CreateFromTask(ViewModel!.OpenNexusModPage);
+            GetModsFromDriveButton.Command = ReactiveCommand.CreateFromTask(
+                async () => await ViewModel!.RegisterFromDisk(TopLevel.GetTopLevel(this)!.StorageProvider));
```

Comment:
Why are these commands in the View? Shouldn't these all be in the ViewModel and shouldn't this be using `BindCommand()` ?